dev-manuals/architecture/overworld/README.md

```diff
@@ -0,0 +1,53 @@
+
+See the README or the user manual.
+
+## What to look at
+
+- [Assets/Scripts/Barriers/*](https://github.com/Gamify-IT/overworld/tree/main/Assets/Scripts/Barriers) (TODO: adjust comments)
+- [Assets/Scripts/GameManager/GameManagerV2.cs](https://github.com/Gamify-IT/overworld/blob/main/Assets/Scripts/GameManager/GameManagerV2.cs) (TODO: adjust comments)
+- [Assets/Scripts/GameManager/GameSettings.cs](https://github.com/Gamify-IT/overworld/blob/main/Assets/Scripts/GameManager/GameSettings.cs) (TODO: adjust comments)
+- [Assets/Scripts/GameManager/JsonHelper.cs](https://github.com/Gamify-IT/overworld/blob/main/Assets/Scripts/GameManager/JsonHelper.cs) (TODO: adjust comments)
+- [Assets/Scripts/GameManager/NPCTalkEvent.cs](https://github.com/Gamify-IT/overworld/blob/main/Assets/Scripts/GameManager/NPCTalkEvent.cs) (TODO: adjust comments)
+- [Assets/Scripts/GameManager/Data Classes/*](https://github.com/Gamify-IT/overworld/blob/main/Assets/Scripts/GameManager/Data%20Classes) (TODO: adjust comments)
+- [Assets/Scripts/GameManager/DTOs/*](https://github.com/Gamify-IT/overworld/blob/main/Assets/Scripts/GameManager/DTOs) (TODO: adjust comments)
+- [Assets/Scripts/HUD/*](https://github.com/Gamify-IT/overworld/blob/main/Assets/Scripts/HUD) (TODO: adjust comments)
+- [Assets/Scripts/Interactable/*](https://github.com/Gamify-IT/overworld/blob/main/Assets/Scripts/Interactable) (TODO: adjust comments, restructure NPC.cs)
+- [Assets/Scripts/MinigameLoading/Configuration.cs](https://github.com/Gamify-IT/overworld/blob/main/Assets/Scripts/MinigameLoading/Configuration.cs) (TODO: adjust comments)
+- [Assets/Scripts/MinigameLoading/Minigame.cs](https://github.com/Gamify-IT/overworld/blob/main/Assets/Scripts/MinigameLoading/Minigame.cs) (TODO: adjust comments)
+- [Assets/Scripts/MinigameLoading/MinigameData.cs](https://github.com/Gamify-IT/overworld/blob/main/Assets/Scripts/MinigameLoading/MinigameData.cs) (TODO: adjust comments)
+- [Assets/Scripts/MinigameLoading/MinigameStarting.cs](https://github.com/Gamify-IT/overworld/blob/main/Assets/Scripts/MinigameLoading/MinigameStarting.cs) (TODO: adjust comments)
+- [Assets/Scripts/Player/Animation.cs](https://github.com/Gamify-IT/overworld/blob/main/Assets/Scripts/Player/Animation.cs) (TODO: adjust comments, rename)
+- [Assets/Scripts/Scene Loading/*](https://github.com/Gamify-IT/overworld/blob/main/Assets/Scene%20Loading) (TODO: adjust comments, remove commented out code, rename loadmaps -> LoadMaps)
+- [Assets/Plugins/LoadMinigame.jslib](https://github.com/Gamify-IT/overworld/blob/main/Assets/Plugins/LoadMinigame.jslib)
+
+## What to ignore
+
+- Assets/Scripts/GameManager/GameManager.cs (TODO: delete)
+- Assets/Scripts/GameManager/EnterArea.cs (TODO: delete)
+- Assets/Scripts/MinigameLoading/LoadMinigame.cs (TODO: delete)
+- Assets/Scripts/Player/mymove.cs (TODO: delete)
+- Assets/Scripts/MoveTilesEditor.cs
+
+## Known Design Flaws
+
+- Unity :,)
+- UniTask return values are not working correctly
```

Comment:
For someone who hasn't worked much with Unity, this says nothing.
What's the implication why this affects us?

Reply:
This is also more of a joke and is still removed

We use UniTasks because normal Unity tasks do not work in WebGL. The advantage of tasks is that they can run asynchronously and return a result. But UniTasks only return their status, as finished or unfinished, which affects us, because we don't get the data that was processed in the task directly from it, but have to query it separately...